### ISSUE TITLE
[Issue] Better error message for forgetting 'azd-service-name' tag

### ISSUE THREAD

**Issue by pamelafox** [Azure/azure-dev#1037]
**Output from `azd version`**

```
azd version 0.3.0-beta.4-pr.1945868 (commit 3aadf6dc4c842ba6813e423e4e9d6788e6dff200)
```

**Output from `az version`** 

```
{
  "azure-cli": "2.40.0",
  "azure-cli-core": "2.40.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "containerapp": "0.3.11",
    "containerapp-compose": "0.2.2"
  }
}
```

**Describe the bug**

I had forgotten the step of tagging the function app resource with a matching service name to the name in azure.yaml ('azd-service-name': 'api').

With that missing, I got an error message about the API endpoint URL:

```
% azd deploy 
Deploying service api...
Error: deploying service: deploying service api package: 
failed executing request: 
Post "https://salary-modelapi.scm.azurewebsites.net/api/zipdeploy?isAsync=true": 
dial tcp: lookup salary-modelapi.scm.azurewebsites.net: no such host
```

It took me a bit of poking around to realize my actual error. Is it possible to detect the lack of tag issue earlier and just say something like "Cannot find a provisioned resource with 'azd-service-name' tag matching the service name 'api' from azure.yaml, please check your Bicep files".

This is not a big deal, but it may be a somewhat common step to forget, especially if someone's reusing Bicep files from non-azd sources like I was.

Here's my diff that fixed it:
https://github.com/pamelafox/regression-model-azure-demo/commit/d975429ef11c9fef37d9599706303964a7e037e2

**To Reproduce**

1. Get very latest azd version (main trunk)
2. git clone https://github.com/pamelafox/regression-model-azure-demo/ @ 72b7c07216caaece9b209d9b08601a7c84da036c 
3. Run azd up

**Environment**
Information on your environment: Mac OSX Ventura, Terminal

**Comment by mikaelweave** [comment]
This same issue occurs if you deploy a functionapp and accidentally tag the app plan with the same `azd-service-name`.  I did a little "refactoring" and accidentally set the tag too high. It resulted in this error even though my functionapp had the tag.

```
Deploying service AzureAuth...
Error: deploying service: deploying service AzureAuth package: failed executing request: Post "https://mikaelw-smart4AzureAuth.scm.azurewebsites.net/api/zipdeploy?isAsync=true": dial tcp: lookup mikaelw-smart4AzureAuth.scm.azurewebsites.net: no such host
```

**Comment by weikanglim** [comment]
Fixed with #1181.

An error message like "unable to find a resource tagged with 'azd-service-name: web'. Ensure the service resource is correctly tagged in your bicep files, and rerun provision" will be displayed in these cases.

**Comment by pamelafox** [comment]
Great, thank you!
